scripts/build.py

```diff
@@ -147,26 +147,46 @@
     '--minify_third_party_libs_only', action='store_true', default=False,
     dest='minify_third_party_libs_only')
 _PARSER.add_argument(
-    '--deparallelize_terser', action='store_true', default=False,
-    dest='deparallelize_terser')
+    '--deparallelize_terser',
+    action='store_true',
+    default=False,
+    dest='deparallelize_terser',
+    help='Disable parallelism on terser plugin in webpack. Use with prod_env.')
+_PARSER.add_argument(
+    '--maintenance_mode',
+    action='store_true',
+    default=False,
+    dest='maintenance_mode',
+    help=(
+        'Enable maintenance mode, '
+        'meaning that only super admins can access the site.'
+    )
+)
 
 
-def generate_app_yaml(deploy_mode=False):
+def generate_app_yaml(deploy_mode=False, maintenance_mode=False):
     """Generate app.yaml from app_dev.yaml.
 
     Args:
-        deploy_mode: bool. Whether the script is being called
-        from deploy script.
+        deploy_mode: bool. Whether the script is being called from deploy
+            script.
+        maintenance_mode: bool. Whether the site should be put into
+            maintenance mode.
     """
     prod_file_prefix = 'build/'
+    maintenance_page_path = 'webpack_bundles/maintenance-page.mainpage.html'
     content = '# THIS FILE IS AUTOGENERATED, DO NOT MODIFY\n'
     with python_utils.open_file(APP_DEV_YAML_FILEPATH, 'r') as yaml_file:
         content += yaml_file.read()
     for file_path in FILEPATHS_NOT_TO_RENAME:
         if 'webpack_bundles/' in file_path:
-            content = content.replace(
-                file_path,
-                prod_file_prefix + file_path)
+            if maintenance_mode:
+                content = content.replace(
+                    file_path, prod_file_prefix + maintenance_page_path)
```

Comment:
Are we going to serve assets (images/videos), js and ts files in maintenance mode?

URLs like: https://www.oppia.org/assets/images/avatar/oppia_avatar_100px.svg https://www.oppia.org/templates/css/oppia.css https://www.oppia.org/third_party/generated/js/third_party.min.js

Reply:
For quick review, just wanted to confirm this is fine either way. The main thing in maintenance mode is that no *changes* can be made to the system.